File: src/main/java/com/cag/cagbackendapi/entities/UnionStatusMemberEntity.java
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "union_status_member")
public class UnionStatusMemberEntity {
    @Id
    @Type(type = "pg-uuid")
    private UUID union_status_member_id;

    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_profile"))
    private ProfileEntity profileEntity;

    @ManyToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_union_status"))
    private UnionStatusEntity unionStatusEntity;
}

File: src/main/java/com/cag/cagbackendapi/entities/UnionStatusMemberEntity.kt
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "union_status_member")
data class UnionStatusMemberEntity (
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    var union_status_member_id: UUID?,

    @ManyToOne
    @JoinColumn(name = "profile_id", referencedColumnName = "profile_id")
    var profileEntity: ProfileEntity?,

    @ManyToOne
    @JoinColumn(name = "union_status_id", referencedColumnName = "union_status_id")
    var unionStatusEntity: UnionStatusEntity?
)
